src/nodes.coffee

```diff
@@ -2610,8 +2610,8 @@ exports.Code = class Code extends Base
         @isGenerator = yes
       if (node instanceof Op and node.isAwait()) or node instanceof AwaitReturn
         @isAsync = yes
-      if @isGenerator and @isAsync
-        node.error "function can't contain both yield and await"
+      if (node instanceof For and node.isAwait())
```

Comment:
This feels a lot less readable than `if @isGenerator and @isAsync`. Why was this changed?

Reply:
Becuase I'm checking if this is async iterator `for await x from y`. 
`if @isGenerator and @isAsync` was used to check if both condition above returned true and throw an error. Since functions can now contain `yield` and `await`  this condition isn't required.
Besides, I think we can't use `@isAsync` here, e.g. `if node instanceof For and @isAwait` here, as `@isAwait` will be false.